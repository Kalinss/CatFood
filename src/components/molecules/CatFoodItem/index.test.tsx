import React from "react";
import { CatFoodItem } from "./index";
import renderer from "react-test-renderer";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const mockDataDefault = {
  id: 0,
  description: {
    value: "Сказочное заморское яство",
    alternative: "Котэ не одобряет?",
  },
  title: "Нямушка",
  subTitle: "c фуа-гра",
  count: {
    value: "0,5",
    metric: "кг",
  },
  info: ["<b>10</b> порций", "мышь в подарок"],
  isActive: false,
  disabled: {
    value: false,
    warn: "Печалька, с фуа-гра закончился.",
  },
  additional: {
    value: "Чего сидишь? Порадуй котэ, <a>купи</a>",
    alternative: "Печень утки разварная с артишоками.",
  },
  fullDescription:
    "Кошачий корм Нямушка 2 киллограмма, это 40 порций, 2 мыши в подарок",
};
const mockDataDisabled = {
  id: 0,
  description: {
    value: "Сказочное заморское яство",
    alternative: "Котэ не одобряет?",
  },
  title: "Нямушка",
  subTitle: "c фуа-гра",
  count: {
    value: "0,5",
    metric: "кг",
  },
  info: ["<b>10</b> порций", "мышь в подарок"],
  isActive: true,
  disabled: {
    value: true,
    warn: "Печалька, с фуа-гра закончился.",
  },
  additional: {
    value: "Чего сидишь? Порадуй котэ, <a>купи</a>",
    alternative: "Печень утки разварная с артишоками.",
  },
  fullDescription:
    "Кошачий корм Нямушка 2 киллограмма, это 40 порций, 2 мыши в подарок",
};
const mockDataActive = {
  id: 0,
  description: {
    value: "Сказочное заморское яство",
    alternative: "Котэ не одобряет?",
  },
  title: "Нямушка",
  subTitle: "c фуа-гра",
  count: {
    value: "0,5",
    metric: "кг",
  },
  info: ["<b>10</b> порций", "мышь в подарок"],
  isActive: true,
  disabled: {
    value: false,
    warn: "Печалька, с фуа-гра закончился.",
  },
  additional: {
    value: "Чего сидишь? Порадуй котэ, <a>купи</a>",
    alternative: "Печень утки разварная с артишоками.",
  },
  fullDescription:
    "Кошачий корм Нямушка 2 киллограмма, это 40 порций, 2 мыши в подарок",
};

describe("CatFoodItem", () => {
  it("should be rendered", () => {
    const tree = renderer
      .create(
        <CatFoodItem iter={0} item={mockDataDefault} clickHandler={() => {}} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be have prop class", () => {
    const wrapper = shallow(
      <CatFoodItem
        item={mockDataDefault}
        clickHandler={() => {}}
        classStyle={"123"}
        iter={0}
      />
    );
    expect(wrapper.hasClass("123")).toBe(true);
  });

  jest.useFakeTimers();

  it("should toggle alternative description info after mouseEnter and mouseLeave event", () => {
    const wrapper = shallow(
      <CatFoodItem iter={0} item={mockDataActive} clickHandler={() => {}} />
    );
    wrapper.find(".item").simulate("mouseEnter");
    jest.advanceTimersByTime(500);
    expect(wrapper.find(".description").text()).toStrictEqual(
      mockDataActive.description.alternative
    );
    wrapper.find(".item").simulate("mouseLeave");
    jest.advanceTimersByTime(500);
    expect(wrapper.find(".description").text()).toStrictEqual(
      mockDataActive.description.value
    );
  });

  it("should showed alternative additional after click", () => {
    const wrapper = shallow(
      <CatFoodItem iter={0} item={mockDataActive} clickHandler={() => {}} />
    );
    wrapper.simulate("click");
    jest.advanceTimersByTime(500);
    expect(wrapper.find(".additional").text()).toStrictEqual(
      mockDataActive.additional.alternative
    );
    wrapper.unmount();
  });

  it("should showed alternative value in additional field, for active component", () => {
    const wrapper = shallow(
      <CatFoodItem iter={0} item={mockDataActive} clickHandler={() => {}} />
    );
    expect(wrapper.find(".additional").text()).toStrictEqual(
      mockDataActive.additional.alternative
    );
  });

  it("should show warn text,if component is disabled", () => {
    const wrapper = shallow(
      <CatFoodItem iter={0} item={mockDataDisabled} clickHandler={() => {}} />
    );
    expect(wrapper.find(".additional").text()).toBe(
      mockDataDisabled.disabled.warn
    );
  });

  it("should have active class", () => {
    const wrapper = shallow(
      <CatFoodItem iter={0} item={mockDataActive} clickHandler={() => {}} />
    );
    expect(wrapper.find(".item").hasClass("active"));
  });
  it("should have disabled class", () => {
    const wrapper = shallow(
      <CatFoodItem iter={0} item={mockDataDisabled} clickHandler={() => {}} />
    );
    expect(wrapper.find(".item").hasClass("disabled"));
  });
});
