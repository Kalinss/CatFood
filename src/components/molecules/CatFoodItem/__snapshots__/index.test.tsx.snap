// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatFoodItem should be rendered 1`] = `
<div
  className="wrapper"
>
  <div
    aria-checked={false}
    className="item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="checkbox"
    tabIndex={1}
  >
    <svg
      className="border svg"
      id="Слой_2"
      viewBox="0 0 320 480"
      x="0px"
      xmlSpace="preserve"
      y="0px"
    >
      <path
        className="path"
        d="M 305 478 H 12.3 C 6.6 478 2 473.4 2 467.7 V 45 c 0 -3 5 -6 10 -11 c 8.5 -8.5 17.4 -17.5 25 -25 c 5 -5 5.1 -7 8 -7 h 254 c 14.5 -0.5 20.5 -0.5 19 17 v 439 C 317.5 472 318 478 305 478 Z"
      />
    </svg>
    <div
      className="main"
    >
      <p
        aria-hidden="true"
        className="description mouseLeave"
      >
        Сказочное заморское яство
      </p>
      <div
        aria-hidden="true"
      >
        <div
          className="header title"
        >
           
          <h2>
            Нямушка
          </h2>
           
        </div>
        <div
          className="header subTitle"
        >
           
          <span>
            c фуа-гра
          </span>
           
        </div>
      </div>
      <div
        aria-hidden="true"
        className="info"
      >
        <ul>
          <li>
            <b>
              10
            </b>
             порций
          </li>
          <li>
            мышь в подарок
          </li>
        </ul>
      </div>
      <div
        aria-hidden="true"
      >
        <div
          className="body price"
        >
          <span
            className="count"
          >
            0,5
          </span>
          <span
            className="metric"
          >
            кг
          </span>
        </div>
      </div>
    </div>
    <div
      className="srOnly"
    >
      <p>
        Кошачий корм Нямушка 2 киллограмма, это 40 порций, 2 мыши в подарок
      </p>
      <p>
        Добавить в корзину нажатием enter
      </p>
      <div
        aria-live="polite"
        role="status"
      >
        <p
          aria-hidden={false}
        >
          Не добавлено
        </p>
        <p
          aria-hidden={true}
        >
          Добавлено
        </p>
        <p
          aria-hidden={true}
        >
          Не добавлено, корм закончился
        </p>
      </div>
    </div>
  </div>
  <p
    aria-hidden="true"
    className="additional"
    onClick={[Function]}
  >
    Чего сидишь? Порадуй котэ, 
    <a>
      купи
    </a>
  </p>
</div>
`;
